src/otx/cli/utils/help_formatter.py

```diff
@@ -0,0 +1,224 @@
+"""Custom Help Formatters for OTX CLI."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import re
+import sys
+from typing import TYPE_CHECKING, Iterable
+
+from jsonargparse import DefaultHelpFormatter
+from rich.markdown import Markdown
+from rich.panel import Panel
+from rich_argparse import RichHelpFormatter
+
+from otx.core.config.engine import EngineConfig
+
+if TYPE_CHECKING:
+    import argparse
+
+    from rich.console import RenderableType
+
+
+BASE_ARGUMENTS = {"config", "print_config", "help", "engine", "model", "model.help", "data_root", "task"}
+ENGINE_ARGUMENTS = {f"engine.{field.name}" for field in EngineConfig.__dataclass_fields__.values()}
+REQUIRED_ARGUMENTS = {
+    "train": {
+        "data",
+        "optimizer",
+        "scheduler",
+    }.union(BASE_ARGUMENTS).union(ENGINE_ARGUMENTS),
+    "test": {
+        "data",
+    }.union(BASE_ARGUMENTS).union(ENGINE_ARGUMENTS),
+}
+
+
+def get_verbosity_subcommand() -> dict:
+    """Parse command line arguments and returns a dictionary of key-value pairs.
+
+    Returns:
+        A dictionary containing the parsed command line arguments.
+
+    Examples:
```

Comment:
Thanks for adding this section. Always makes it easy to follow :)
```suggestion
    Example:
```